core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -292,13 +292,20 @@ private boolean rebalancePartition(WriterId from, WriterId to)
             return false;
         }
 
-        private void updateWriterEstimatedWrittenBytes(WriterId from, WriterId to, long estimatedPartitionWrittenBytesSinceLastRebalance, int newWriterCount)
+        private void updateAffectedWriterandEstimatedWrittenBytes(WriterId to, long estimatedPartitionWrittenBytesSinceLastRebalance, PartitionInfo partitionInfo)
         {
             // Since a partition is rebalanced from max to min skewed writer, decrease the priority of max
             // writer as well as increase the priority of min writer.
+            int newWriterCount = partitionInfo.getWriterCount();
             int oldWriterCount = newWriterCount - 1;
-            writerEstimatedWrittenBytes[from.id] -= (estimatedPartitionWrittenBytesSinceLastRebalance * oldWriterCount) / newWriterCount;
-            writerEstimatedWrittenBytes[to.id] += (estimatedPartitionWrittenBytesSinceLastRebalance * oldWriterCount) / newWriterCount;
+            for (int writer : partitionInfo.getWriterIds()) {
+                if (writer != to.id) {
+                    writerEstimatedWrittenBytes[writer] -= estimatedPartitionWrittenBytesSinceLastRebalance / newWriterCount;
```

Comment:
nit: technically we probably should compute averaged `estimatedPartitionWrittenBytesSinceLastRebalance` from all writers
OR
compute `estimatedPartitionWrittenBytesSinceLastRebalance` per writer.

But I think it's fine for now

Reply:
I agree with you, it’s just an estimation，maybe there is not too much need of accuracy.

Reply:
Ideally `TableWriter` operator provides physical written bytes per partition, so no estimation is needed. This would require moving of `pageIndexer` logic to trino-main from connectors, which is much more involving change.